hedera-node/src/main/java/com/hedera/services/ledger/TransactionalLedger.java

```diff
@@ -508,6 +514,20 @@ private boolean isZombie(K id) {
 		return prop -> prop.getter().apply(defaultEntity);
 	}
 
+	private List<MerkleLeafChanges<K, A, P>> getAccumulatedMerkleLeafChanges() {
+		final List<MerkleLeafChanges<K, A, P>> merkleLeavesChanges = new ArrayList<>();
+		for (final var merkleLeafChangesEntry : changes.entrySet()) {
+			final var id = merkleLeafChangesEntry.getKey();
+			final var merkleProperties = merkleLeafChangesEntry.getValue();
+			final var merkleLeaf = entities.contains(id) ? entities.getRef(id) :
+					newEntity.get();
```

Comment:
I'd suggest using `null` to represent there was no existing leaf (i.e. this change is a creation).